compiler/cli/src/org/jetbrains/kotlin/cli/jvm/compiler/CompileEnvironmentUtil.java

```diff
@@ -125,21 +129,30 @@ private static void writeRuntimeToJar(JarOutputStream stream, boolean resetJarTi
         copyJarImpl(stream, stdlibPath, resetJarTimestamps);
     }
 
+    private static void writeReflectToJar(JarOutputStream stream, boolean resetJarTimestamps) throws IOException {
+        File reflectPath = PathUtil.getKotlinPathsForCompiler().getReflectPath();
+        if (!reflectPath.exists()) {
+            throw new CompileEnvironmentException("Couldn't find kotlin-reflect at " + reflectPath);
+        }
+        copyJarImpl(stream, reflectPath, resetJarTimestamps);
+    }
+
     private static void copyJarImpl(JarOutputStream stream, File jarPath, boolean resetJarTimestamps) throws IOException {
         try (JarInputStream jis = new JarInputStream(new FileInputStream(jarPath))) {
             while (true) {
                 JarEntry e = jis.getNextJarEntry();
                 if (e == null) break;
-
-                if (!FileUtilRt.extensionEquals(e.getName(), "class") ||
-                    StringsKt.substringAfterLast(e.getName(), "/", e.getName()).equals("module-info.class")) {
+                if (StringsKt.substringAfterLast(e.getName(), "/", e.getName()).equals("module-info.class")) {
                     continue;
                 }
                 if (resetJarTimestamps) {
                     e.setTime(DOS_EPOCH);
                 }
-                stream.putNextEntry(e);
-                FileUtil.copy(jis, stream);
+                if (FileUtilRt.extensionEquals(e.getName(), "class") ||
+                    FileUtilRt.extensionEquals(e.getName(), "kotlin_builtins")) {
+                    stream.putNextEntry(e);
+                    FileUtil.copy(jis, stream);
+                }
```

Comment:
Some changes around here, are used  to avoid complicating the conditional statements.

Reply:
IMHO it'd be better to split the `if` above, to make it clearer that we don't call `e.setTime` for entries which are not class files / kotlin_builtins files

Reply:
Thanks for let me know that, I didn't realize it. I have reverted these improper changes.